tf_encrypted/protocol/pond.py

```diff
@@ -989,6 +989,77 @@ def dispatch(self, base_name, *args, container=None, **kwargs):
         else:
             raise TypeError("Don't know how to {}: {}".format(base_name, [type(arg) for arg in args]))
 
+    @memoize
+    def zeros(
+        self,
+        shape,
+        tensor_type: str = "private",
+        apply_scaling: bool = True,
+        name: Optional[str] = None,
+        factory: Optional[AbstractFactory] = None
+    ) -> Union['PondPublicTensor', 'PondPrivateTensor', 'PondMaskedTensor']:
+
+        if tensor_type == "public":
+            out = _zeros_public(self, shape, apply_scaling, name, factory)
+        elif tensor_type == "private":
+            out = _zeros_private(self, shape, apply_scaling, name, factory)
+        elif tensor_type == "masked":
+            out = _zeros_masked(self, shape, apply_scaling, name, factory)
+
+        return out
+
+    def pad(self, x: 'PondTensor', paddings: list):
+
+        with tf.name_scope('pad'):
+            zeros_id = 0
+            for axis, (pad_before, pad_after) in enumerate(paddings):
+                x = self._append_zeros(x, pad_after, axis, zeros_id)
+                zeros_id += 1
+                x = self._prepend_zeros(x, pad_before, axis, zeros_id)
+                zeros_id += 1
+
+        return x
+
+    def _prepend_zeros(self, arr: np.ndarray, pad_amt: int, axis: int, zeros_id: int):
+
+        if pad_amt == 0:
+                return arr
+
+        arrshape = arr.shape.as_list()
+        padshape = tuple(x if i != axis else pad_amt for (i, x) in enumerate(arrshape))
+
+        if isinstance(arr, PondPublicTensor):
```

Comment:
related to above, you shouldn't need to check the instances and pass a specific tensor type, it should just pick the right one automatically if you have the base zeros set up correctly